src/core/net/ip4_address.hpp

```diff
@@ -101,12 +138,76 @@ class Address : public Equatable<Address>, public Clearable<Address>
      *
      */
     InfoString ToString(void) const;
-
-private:
-    uint8_t mBytes[kSize];
 } OT_TOOL_PACKED_END;
 
+class Cidr : public otIp4Cidr, public Unequatable<Cidr>, public Clearable<Address>
+{
+public:
+    static constexpr uint16_t kCidrSuffixSize = 3; ///< Suffix to represent CIDR (/dd).
+
+    /**
+     * This type defines the fixed-length `String` object returned from `ToString()`.
+     *
+     */
+    typedef String<Address::kAddressStringSize + kCidrSuffixSize> InfoString;
+
+    /**
+     * This method converts the IPv4 address to a string.
+     *
+     * The string format uses quad-dotted notation of four bytes in the address (e.g., "127.0.0.1").
+     *
+     * @returns An `InfoString` representing the IPv4 address.
+     *
+     */
+    InfoString ToString(void) const;
+
+    /**
+     * This method returns the host mask (bitwise not of the subnet mask) of the CIDR.
+     *
+     * @returns A uint32 for the host mask, in network byte order.
+     */
+    uint32_t HostMask(void) const { return HostSwap32((uint32_t(1) << uint32_t(32 - mLength)) - 1); }
```

Comment:
I think we may not need the cast to `uint32_t` in  `uint32_t(32 - mLength)`.
Also can use `1UL` instead of `uint32_t(1)`

Reply:
After facing the int width issue, I found a more elegant way is just `0xffffffff >> mLength`